### PR TITLE
stolostron/backlog#21395 submar openstack

### PR DESCRIPTION
**Pull request by cadawson**

### CODE REVIEW

File: services/submariner/submariner_prepare_hosts.adoc [hunk 1]
```diff
@@ -234,6 +234,73 @@ Submariner uses IP security (IPsec) to establish the secure tunnels between the
 [#preparing-openstack]
 == Preparing Red Hat OpenStack Platform for Submariner
 
-To prepare Red Hat OpenStack Platform clusters for deploying Submariner, complete the following steps:
+You can use the `SubmarinerConfig` API to configure the Red Hat OpenStack Platform cluster to integrate with a Submariner deployment. To prepare Red Hat OpenStack Platform clusters for deploying Submariner, complete the following steps:
 
-...
\ No newline at end of file
+. If you did not create the managed cluster with {product-title-short}, you must manually create a secret on your hub cluster in the namespace of your managed cluster that contains your Red Hat OpenStack Platform credential secret. If you created the cluster with {product-title-short}, then skip to step 2.
```
Comment (aswinsuryan) [inline services/submariner/submariner_prepare_hosts.adoc]: In either case we need to create the secret manually. As the secret used by submariner has different requirements from the ACM one.

File: services/submariner/submariner_prepare_hosts.adoc [hunk 2]
```diff
@@ -234,6 +234,73 @@ Submariner uses IP security (IPsec) to establish the secure tunnels between the
...
+   clouds.yaml: <rhos-clouds-yaml-file-content>
+   cloud : <rhos-cloud-name>
+auth:
+   auth_url: https://rhos-d.infra.prod.upshift.rdu2.redhat.com:13000
+   application_credential_id: <id>
+   application_credential_secret: <secret>
+region_name: "regionOne"
+interface: "public"
+identity_api_version: 3
+auth_type: "v3applicationcredential"
```
Comment (aswinsuryan) [inline services/submariner/submariner_prepare_hosts.adoc:260]: The credential secret required by submariners takes these parameters.

    auth:
      auth_url: https://rhos-d.infra.prod.upshift.rdu2.redhat.com:13000
      application_credential_id: "<id>"
      application_credential_secret: "<secret>"
    region_name: "regionOne"
    interface: "public"
    identity_api_version: 3
    auth_type: "v3applicationcredential"

File: add-ons/submariner/submariner_prepare_hosts.adoc [hunk 3]
```diff
@@ -234,6 +234,73 @@ Submariner uses IP security (IPsec) to establish the secure tunnels between the
...
+apiVersion: v1
+kind: Secret
+metadata:
+   name: <managed-cluster-name>-rhos-creds
+   namespace: <managed-cluster-namespace>
+type: Opaque
+data:
+   clouds.yaml: <rhos-clouds-yaml-file-content>
+   cloud : <rhos-cloud-name>
+auth:
```
Comment (aswinsuryan) [inline add-ons/submariner/submariner_prepare_hosts.adoc]: oh I missed this , the auth section is a part of clouds,yaml , the secret should be

apiVersion: v1
data:
  cloud: <cloudname>
  clouds.yaml:<clouds.yaml>
kind: Secret
metadata:
  name: <name>
  namespace: <namespace>
type: Opaque

and the clouds.yaml above should be base 64 encoded of 

clouds:
  openstack:
    auth:
      auth_url: <auth-url>
      application_credential_id: <application-credential-id>
      application_credential_secret: <application-credential-secret>
    region_name: "regionOne"
    interface: "public"
    identity_api_version: 3
    auth_type: "v3applicationcredential"

**Review by aswinsuryan — CHANGES_REQUESTED** [review]

File: add-ons/submariner/submariner_prepare_hosts.adoc (hunk 3 above)
Comment (cadawson) [reply add-ons/submariner/submariner_prepare_hosts.adoc]: I think I have this covered now, @aswinsuryan . Please take a look. Thanks!

**Comment by cadawson** [comment]
Is this one ready to merge @aswinsuryan ? Thanks!

File: add-ons/submariner/submariner_prepare_hosts.adoc [hunk 4]
```diff
@@ -234,6 +234,83 @@ Submariner uses IP security (IPsec) to establish the secure tunnels between the
...
+Replace `managed-cluster-namespace` with the namespace of your managed cluster.
++
+Replace `managed-cluster-name` with the name of your managed cluster. The value of `managed-cluster-name-rhos-creds` is your Red Hat OpenStack Platform credential secret name, which you can find in the cluster namespace of your hub cluster.
++
+*Note:* The name of the `SubmarinerConfig` custom resource must be `submariner`, as shown in the example.
+
+.. Run the following command to apply the file:
++
+----
+oc apply -f submar_addon
```
Comment (vthapar) [inline add-ons/submariner/submariner_prepare_hosts.adoc]:
```suggestion
oc apply -f submar_addon.yaml
```

File: add-ons/submariner/submariner_prepare_hosts.adoc [hunk 5]
```diff
@@ -234,6 +234,83 @@ Submariner uses IP security (IPsec) to establish the secure tunnels between the
...
 == Preparing Red Hat OpenStack Platform for Submariner
 
-To prepare Red Hat OpenStack Platform clusters for deploying Submariner, complete the following steps:
+You can use the `SubmarinerConfig` API to configure the Red Hat OpenStack Platform cluster to integrate with a Submariner deployment. To prepare Red Hat OpenStack Platform clusters for deploying Submariner, complete the following steps:
 
-...
\ No newline at end of file
+. Create a base 64-encoded file titled `clouds.yaml` with the authentication information for your Red Hat OpenStack Platform environment. The file should resemble the following example: 
++
+[source.yaml]
```
Comment (vthapar) [inline add-ons/submariner/submariner_prepare_hosts.adoc]: Shouldn't this be [source,yaml] ?

**Review by vthapar — CHANGES_REQUESTED** [review]

File: add-ons/submariner/submariner_prepare_hosts.adoc (hunk 5 above)
Comment (cadawson) [reply add-ons/submariner/submariner_prepare_hosts.adoc]: Yes. Good catch. Changed.

**Review by aswinsuryan — APPROVED** [review]

**Pull request merged**